src/components/SearchModal/CurrencySearch.tsx

```diff
@@ -70,6 +73,14 @@ export function CurrencySearch({
   const searchToken = useToken(debouncedQuery)
   const searchTokenIsAdded = useIsUserAddedToken(searchToken)
 
+  // TO-DO find out why non-conditional usage of this hook blocks the UI thread
+  const searchTokenWithRestrictions = searchToken
```

Comment:
@ra-phael Got any ideas why calling this hook unconditionally blocks up the UI thread? Seems to fire something continuously which halts everything else.

Tried a bunch of debugging but not sure what's wrong! Please take a look🙏